app/main/views/direct_award.py

```diff
@@ -85,6 +97,7 @@ def create_project():
         user=updater_json['updated_by'],
         data={
             'projectId': project.id,
+            'projectExternalId': project.external_id,
```

Comment:
This is just creating an AuditEvent, which is still fairly internal to the API. It's probably good to have both references because you could end up with either one (internal from some error in the API, external from a user), so allows you to search audit events with either reference.

This implementation isn't really about having transparent external IDs any more. It's now something developers need to know about and understand as part of how we reference resources - internal IDs for eg database joins, but external IDs as soon as it leaves the API/DB.

`project.id` should never be emitted in a response by the API, but it will still get used internally.

Reply:
Ah OK, sorry yeah missed that. But in that case isn't the `projectId` the same as the `objectId` on the audit event, in which case we shouldn't need to capture it in the `data` field? And then we can look up the external idfrom the object id  where necessary, as we do for service IDs I think?  Although maybe it is nicer not to have to do that lookup...

Reply:
Hmm, looks like you're right, we have object_id directly and that is indeed the project's internal ID. So you think we should retain the `projectId` key in the AuditEvent's data and populate it with the project's external ID?

Reply:
No, I think the object ID for audit events should probably always reference the internal ID.  At least, this is what we do for audit events relating to services.  Which might not be the best thing, but now we've got it let's stay consistent.

For services we don't even capture the external service id in the data at all.

But feels like it could be convenient to do so, so maybe keep the `external` in data here?

Reply:
Sorry, I wasn't clear. `object_id` should definitely stay as the internal ID. I was asking if we should change projectId to reference the external ID or if we should add a new key to the data for projectExternalId.... as this change is.

Basically do you want me to change anything here?

Reply:
The change here puts two Id's in the JSON data field of the audit event. We need at most one in there (the external ID). The internal project ID is already stored as the object ID on the audit event, so definitely doesn't need to go in the data.

I'm in two minds about having the external one in there, as we don't do that for services AFAIK (although come to think of it maybe we do in the service creation event? Definitely not for other service events). But it feels like it could be useful and no harm to having it in here, so maybe keep it.

Reply:
The change here puts two Id's in the JSON data field of the audit event. We need at most one in there (the external ID). The internal project ID is already stored as the object ID on the audit event, so definitely doesn't need to go in the data.

I'm in two minds about having the external one in there, as we don't do that for services AFAIK (although come to think of it maybe we do in the service creation event? Definitely not for other service events). But it feels like it could be useful and no harm to having it in here, so maybe keep it.